### ISSUE TITLE
FileVault Escrow encryption status unreliable...

### ISSUE THREAD

**Issue by childrss** [munkireport/munkireport-php#1052]
While the display for the FileValut Escrow page is highly useful -- it shows exactly what we need in the field as technicians AND what we need to turn into our Office of Security and Privacy when a machine is lost/stolen -- we've found that the data on whether a drive is encrypted is unreliable.

Looking to the future, as a field technician I would like to have everything I need to access a machine for troubleshooting on one page:  FileVault Escrow, FirmWare Escrow, and LAPS (local admin password).  MunkiReport is really close there and is by far the most useful tool I've seen for front-line techs, middle managment and upper managment.

So the layout AND the way the information is displayed (especially the "encryption verified as of..." date for turning into the Security & Privacy Office) is stellar.  But wrong information is worse than no information.  And in this case FileVault Escrow encryption is often wrong.

**Comment by rickheil** [comment]
Can you provide some more details about the machines when the status is incorrect? Things like the operating system version, MR version, and HFS+ or APFS would be helpful. One other thing - can you verify whether or not the disk shows properly on the Storage tab?

If the storage tab matches 100% of the time, I think the issue here may be that the FileVault status widget is currently just looking at whether or not any data is returned from filevault_status/get_data to determine encryption status rather than looking at the disks table, which is actually querying on the disks themselves. It may make sense to update the logic on the FileVault escrow page to look at the disk report table for current status, or have an "Am I still encrypted?" check built in to the filevault_status module's script.

**Comment by childrss** [comment]
TL; DR:   I think you're spot-on the analysis, in looking at the simple client-side pre-flight code that is only checking for FileVault enabled users and making assumptions based on the result returned for FV enabled users that aren't true 100% of the time.

I've not delved into the server-side code at all, and I'm not a webdev.

Apologies for not including this info, and thanks for the response:

MunkiReport Version 2.15.2.2795, 
The one I'm noticing the issue on *now* on 10.13.4, AFPS iMac (Retina 5K, 27-inch, 2017).   I say "now" because it had an issue where it couldn't complete the update that we've found is most easily fixed by booting to recovery mode and reselecting the Startup Disk.   For encrypted machines this includes mounting the disk in Disk Utility first (and unlocking).

For some reason (not sure why) when I mounted it in Disk Utility when the machine booted it started decrypting the disk.  Not something I recall asking it to do, but I could've slipped up from my normal procedure.   Still on 10.13.3, wouldn't allow upgrade to 10.13.4 but over the weekend the user notified me that it went to 10.13.4.   

In the FileVault escrow tab it says that it is encrypted and has the old escrow key (makes sense, a new key hasn't been recreated & uploaded).  But it also has the status as encrypted and a date it happened, AND misleadingly a date that it was verified encrypted....  I don't believe that it is pulling it from the disk report table.

**Comment by rickheil** [comment]
Got it - thanks for the extra detail. This is exactly the case I was imagining - a machine checks in with a key, and then is decrypted later. I'll mull it over and talk on the slack channel with Bochoven and see what the best solution might be - whether pulling status from disk_report or making FileVault status do more checks itself.

**Comment by ofirgalcon** [comment]
I'm seeing this inconsistency too. I had a separate problem with my crypt settings that was confusing the issue, but now I can see computers sometimes report as encrypted when they are not.

**Comment by nathanperkins** [comment]
It looks like this is probably fixed in the updates that @tuxudo has made to the filevault_status module.

https://github.com/tuxudo/filevault_status/tree/more

Is that something that could be merged?

**Comment by tuxudo** [comment]
The `filevault_status` module has been merged and updated.
